version: '3.8'

services:
  server:
    build: ./server
    command: python3 ./main.py
    environment:
      - DB_USER=${DB_USER}
      - DB_PASSWORD=${DB_PASSWORD}
      - DB_NAME=${DB_NAME}
      - DB_HOST=db
      - DB_PORT=${DB_PORT}
    ports:
      - 5000:5000
    depends_on:
      db:
        condition: service_healthy
    networks:
      - shared_network

  db:
    image: postgres:17.0
    container_name: postgres_container
    restart: always
    environment:
      POSTGRES_USER: ${DB_USER}
      POSTGRES_PASSWORD: ${DB_PASSWORD}
      POSTGRES_DB: ${DB_NAME}
    healthcheck:
      test: ["CMD-SHELL", "pg_isready -U $${POSTGRES_USER} -d $${POSTGRES_DB}"]
      interval: 2s
      timeout: 1s
      retries: 5
    ports:
      - "5432:5432"
    volumes:
      - db_postgres_data-cloned:/var/lib/postgresql/data
    networks:
      - shared_network

networks:
  shared_network:
    driver: bridge

volumes:
  db_postgres_data-cloned:
